lib/src/dialog/types.ts

```diff
@@ -0,0 +1,44 @@
+type Space = "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | "xxlarge";
+type Padding = {
+  top?: Space;
+  bottom?: Space;
+  left?: Space;
+  right?: Space;
+};
+
+type Props = {
+  /**
+   * If true, the close 'x' button will be visible.
+   */
+  isCloseVisible?: boolean;
+  /**
+   * This function will be called when the user clicks the close 'x' button.
+   * The responsibility of hiding the modal lies with the user.
+   */
+  onCloseClick?: () => void;
+  /**
+   * If true, the dialog will be displayed over a darker background that covers the content behind.
+   */
+  overlay?: boolean;
+  /**
+   * This function will be called when the user clicks background of the modal.
+   * The responsibility of hiding the modal lies with the user.
+   */
+  onBackgroundClick?: () => void;
+  /**
+   * Size of the padding to be applied to the component ('xxsmall' | 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge' | 'xxlarge'). 
```

Comment:
Try to remove the values from the comments since they are already defined in the type.

Reply:
Should we not add the possible values in similar cases? 
I was using the button as a template and it has the possible values in the comments.

Reply:
@Jialecl is right. We should add the possible values when the type doesn't explicitly indicate them. This usually happens with Margin type.

Reply:
Why do the possible values have to be indicated if they are already defined in the type? I think it is redundant to repeat this information.

Reply:
I think it's helpful to be able to see the possible values while hovering over the parameter